examples-kotlin/src/main/kotlin/org/bitcoinj/ForwardingService.kt

```diff
@@ -114,72 +125,61 @@ class ForwardingService(val config: Config, val wallet: Wallet) : Closeable {
     }
 
     /**
-     * Forward an incoming transaction by creating a new transaction, signing, and sending to the specified address.
+     * Create a transaction specified by a {@link org.bitcoinj.examples.SendRequest}, sign it, and send to the specified address.
      * @param wallet The active wallet
-     * @param incomingTx the received transaction
-     * @param forwardingAddress the address to send to
+     * @param sendRequest requested transaction parameters
      * @return A future for a TransactionBroadcast object that completes when relay is acknowledged by peers
      */
-    fun forward(
-        wallet: Wallet,
-        incomingTx: Transaction,
-        forwardingAddress: Address?
-    ): CompletableFuture<TransactionBroadcast?> {
-        // Send coins received in incomingTx onwards by sending exactly the outputs that have been sent to us
-        val sendRequest = SendRequest.emptyWallet(forwardingAddress)
-        sendRequest.coinSelector = forwardingCoinSelector(incomingTx.txId)
-        println("Creating outgoing transaction for $forwardingAddress...\n")
+    open fun send(wallet: Wallet, sendRequest: SendRequest?): CompletableFuture<TransactionBroadcast?>? {
```

Comment:
I get this compiler warning:

> 'open' has no effect in a final class

Is it intended? Shall we remove the "open"?

Reply:
yes I can take "open" out. open allows classes and functions to be inherited or overridden. This is not necessary here